scripts/install-windows-prereqs.ps1

```diff
@@ -582,17 +602,29 @@ function Install-DCAP-Dependencies {
                     Throw "Failed searching for $driver driver"
                 }
                 $output | ForEach-Object {
-                    if($_.Contains($drivers[${OS_VERSION}][$driver]['description'])) {
+                    if(($_.Contains($drivers[${OS_VERSION}][$driver]['description'])) -and ($OS_VERSION -eq 'WinServer2016')) {
                         Write-Output "Removing driver $($drivers[${OS_VERSION}][$driver]['location'])"
                         Remove-DCAPDriver -Name $drivers[${OS_VERSION}][$driver]['location']
                     }
                 }
-                Write-Output "Installing driver $($drivers[${OS_VERSION}][$driver]['location'])"
-                $install = & $devConBinaryPath install "$($inf.FullName)" $drivers[${OS_VERSION}][$driver]['location']
-                if($LASTEXITCODE) {
-                    Throw "Failed to install $driver driver"
+                try
+                {
+                    Write-Output "Installing driver $($drivers[${OS_VERSION}][$driver]['location'])"
+                    $install = & $devConBinaryPath install "$($inf.FullName)" $drivers[${OS_VERSION}][$driver]['location']
+                    Write-Output $install
+                    if($LASTEXITCODE) {
+                        Throw "Failed to install $driver driver"
+                    }
+                }
+                catch
```

Comment:
This catch fallback logic doesn't address my concern around the Win10 configuration mentioned in #3230 and it also obscures the intent of the code flow:

* The update logic still references the update target as indexed by `OS_VERSION` so it would fail on Win10 as that configuration did not specify the `hardwareId` values. Was this tested on Win10?
* This `catch` logic looks only ever intended to run for non-WS2016 code (the WS2016 configuration also doesn't specify `hardwareId` values for this codepath) but the way it does this is non-obvious because only WS2016 tries to uninstall a driver first, and is a based on the undocumented expectation that WS2016 does not have a driver installed by default but will uninstall & reinstall over one if it does (as opposed to upgrade).

Also, have we validated that the codepath for _installing_ the WS2019 driver works? I don't think it does because devcon.exe takes the hardware ID in for both the install and update commands; the distinction between the `location` and `hardwareId` parameters defined by this script seems to be spurious and they can probably be consolidated into the more accurately named `hardwareId`.

If uninstall/install is intended to only run for WS2016, can we explicitly break out that behavior by `OE_VERSION` check? Alternatively, if the desire is to always try the install and then try to upgrade, can this be tested to ensure that it works in all configurations (3 OS versions * with/without driver already installed)